### ISSUE TITLE
CI: v1.9/v1.10/v1.11: runtime job is broken

### ISSUE THREAD

**Issue by joestringer** [cilium/cilium#18919]
PR #18892 broke the runtime job:

```
11:35:16      runtime: Feb 23 19:35:14 runtime cilium-agent[21324]: level=info msg="Cilium 1.9.12 cc7220091f 2022-02-23T02:21:27+00:00 go version go1.16.5 linux/amd64" subsys=daemon
11:35:16      runtime: Feb 23 19:35:14 runtime cilium-agent[21324]: level=fatal msg="Envoy: Binary \"cilium-envoy\" cannot be executed" error="exit status 1" subsys=envoy-manager
```

**Comment by jrajahalme** [comment]
I tested this PR locally on a dev VM (net-next) with `test/envoy/envoy-smoke-test.sh` from master and it runs successfully.

**Comment by joestringer** [comment]
My guess is that something is different about the way that the CI provisions & sets up Cilium vs. how the dev VM works. Here's the failure log fwiw:

https://jenkins.cilium.io/job/Cilium-PR-Runtime-4.9/5693/execution/node/92/log/

**Comment by pchaigno** [comment]
Same issue affects v1.10: https://jenkins.cilium.io/job/Cilium-PR-Runtime-net-next/1452/console (PR is https://github.com/cilium/cilium/pull/18835).

@jrajahalme It passed in your case because you used the VM image from master I believe.

Should we attempt to update the VM images we use in v1.9/v1.10 to fix this?

**Comment by nbusseneau** [comment]
> My guess is that something is different about the way that the CI provisions & sets up Cilium vs. how the dev VM works.

If need be, CI VMs can be started locally: https://docs.cilium.io/en/v1.11/contributing/development/dev_setup/#launch-ci-vms

**Comment by joestringer** [comment]
Is it just a matter of bumping the box images like #19047?

**Comment by gandro** [comment]
v1.11 is also affected apparently: https://github.com/cilium/cilium/pull/18905#issuecomment-1061543830

**Comment by joestringer** [comment]
> Is it just a matter of bumping the box images like ~#19047~ #19048?

No, it'll require more effort than that. The latest VM has several missing systemd unit files, it compiles with Go 1.17 which is not consistent for all of the earlier branches, and the latest failure on that branch is some VM swap issue. I'm dropping the PR for now for lack of cycles.

**Comment by jrajahalme** [comment]
Backports that fix this are good to merge, IMO: #19369, #19370, #19373
